## TensorFlow 2.x 학습
> 
> #### [TensorFlow 튜토리얼][TF2_001]
> #### [텐서플로 2.0 시작하기: 초보자용][TF2_002]
> #### [전문가를 위한 TensorFlow 2 빠른 시작][TF2_003]

[TF2_001]: https://www.tensorflow.org/tutorials?hl=ko      "Go TF2_001"
[TF2_002]: https://www.tensorflow.org/tutorials/quickstart/beginner?hl=ko   "Go TF2_002"
[TF2_003]: https://www.tensorflow.org/tutorials/quickstart/advanced?hl=ko   "Go TF2_003"

In [2]:
import tensorflow as tf

#### MNIST 데이터셋을 로드하여 준비
- 샘플 값을 정수에서 부동소수로 변환합니다:

In [3]:

mnist = tf.keras.datasets.mnist

(x_train, y_train), (x_test, y_test) = mnist.load_data()
x_train, x_test = x_train / 255.0, x_test / 255.0

#### 층을 차례대로 쌓아 tf.keras.Sequential 모델을 만듭니다. 
- 훈련에 사용할 옵티마이저(optimizer)와 손실 함수를 선택합니다:

In [4]:
model = tf.keras.models.Sequential([
    tf.keras.layers.Flatten(input_shape=(28, 28)),
    tf.keras.layers.Dense(128, activation='relu'),
    tf.keras.layers.Dropout(0.2),
    tf.keras.layers.Dense(10, activation='softmax')
])

model.compile(optimizer='adam',
              loss='sparse_categorical_crossentropy',
              metrics=['accuracy'])

#### 모델을 훈련하고 평가

In [5]:
model.fit(x_train, y_train, epochs=5)

model.evaluate(x_test,  y_test, verbose=2)

Epoch 1/5
1875/1875 [==============================] - 3s 1ms/step - loss: 0.2935 - accuracy: 0.9139
Epoch 2/5
1875/1875 [==============================] - 2s 1ms/step - loss: 0.1434 - accuracy: 0.9577
Epoch 3/5
1875/1875 [==============================] - 2s 1ms/step - loss: 0.1081 - accuracy: 0.9675
Epoch 4/5
1875/1875 [==============================] - 2s 1ms/step - loss: 0.0878 - accuracy: 0.9726
Epoch 5/5
1875/1875 [==============================] - 3s 1ms/step - loss: 0.0752 - accuracy: 0.9764
313/313 - 0s - loss: 0.0707 - accuracy: 0.9772 - 391ms/epoch - 1ms/step


[0.07067284733057022, 0.9771999716758728]

#### 훈련된 이미지 분류기는 이 데이터셋에서 약 98%의 정확도를 달성합니다. 
- 더 자세한 내용은 TensorFlow 튜토리얼을 참고하세요.